From f7aa866e23f3a8c711a7a2d3a65d90a4655d8882 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 26 Jan 2024 12:12:52 +0100
Subject: [PATCH 0567/1695] dts: stm32: add backup registers configuration for
 stm32mp15x boards

Add a backup register configuration for STM32MP15x boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 37c729753b9a83e6fea00be6c90dd71377afe8a9 dts: stm32: add backup registers configuration for stm32mp15x boards]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I81ce11a18b5b80e9e2bf25e571050170fe3a4bd1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/354735
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp157a-ed1.dts  | 4 ++++
 core/arch/arm/dts/stm32mp157c-ed1.dts  | 4 ++++
 core/arch/arm/dts/stm32mp157d-ed1.dts  | 4 ++++
 core/arch/arm/dts/stm32mp157f-ed1.dts  | 4 ++++
 core/arch/arm/dts/stm32mp15xx-dkx.dtsi | 1 +
 5 files changed, 17 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp157a-ed1.dts b/core/arch/arm/dts/stm32mp157a-ed1.dts
index 7cf554dc3..dcbc035a2 100644
--- a/core/arch/arm/dts/stm32mp157a-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157a-ed1.dts
@@ -353,6 +353,10 @@
 	status = "okay";
 };
 
+&tamp {
+	st,backup-zones = <10 5 17>;
+};
+
 &timers15 {
 	status = "okay";
 
diff --git a/core/arch/arm/dts/stm32mp157c-ed1.dts b/core/arch/arm/dts/stm32mp157c-ed1.dts
index 91bde5290..1daeab5bf 100644
--- a/core/arch/arm/dts/stm32mp157c-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157c-ed1.dts
@@ -543,6 +543,10 @@
 	status = "disabled";
 };
 
+&tamp {
+	st,backup-zones = <10 5 17>;
+};
+
 &timers6 {
 	status = "disabled";
 	/* spare dmas for other usage */
diff --git a/core/arch/arm/dts/stm32mp157d-ed1.dts b/core/arch/arm/dts/stm32mp157d-ed1.dts
index c657ce879..da905e9f1 100644
--- a/core/arch/arm/dts/stm32mp157d-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157d-ed1.dts
@@ -352,6 +352,10 @@
 	status = "okay";
 };
 
+&tamp {
+	st,backup-zones = <10 5 17>;
+};
+
 &timers15 {
 	status = "okay";
 
diff --git a/core/arch/arm/dts/stm32mp157f-ed1.dts b/core/arch/arm/dts/stm32mp157f-ed1.dts
index e3f443d87..d4fd99a84 100644
--- a/core/arch/arm/dts/stm32mp157f-ed1.dts
+++ b/core/arch/arm/dts/stm32mp157f-ed1.dts
@@ -355,6 +355,10 @@
 	status = "okay";
 };
 
+&tamp {
+	st,backup-zones = <10 5 17>;
+};
+
 &timers15 {
 	status = "okay";
 
diff --git a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
index ea4b3b390..d986401c7 100644
--- a/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
+++ b/core/arch/arm/dts/stm32mp15xx-dkx.dtsi
@@ -716,6 +716,7 @@
 
 &tamp {
 	status = "okay";
+	st,backup-zones = <10 5 17>;
 	st,tamp-passive-nb-sample = <4>;
 	st,tamp-passive-sample-clk-div = <16384>;
 };
-- 
2.47.2

